### PR TITLE
Always initialize memory used by kind

### PR DESCRIPTION
**Pull request by bratpiorka**
### Types of changes

- [ ] Bugfix (non-breaking change which fixes issue linked in Description above)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, appearance)
- [ ] Other

### Checklist

- [ ] Code compiles without errors
- [ ] All tests pass locally with my changes (see TESTING section in CONTRIBUTING file)
- [ ] Created tests which will fail without the change (if possible)
- [ ] Extended the README/documentation (if necessary)
- [ ] All newly added files have proprietary license (if necessary)
- [ ] All newly added files are referenced in MANIFEST files (if necessary)

## Further comments

---
This change is [Reviewable](https://reviewable.io/reviews/memkind/memkind/754)

### CODE REVIEW

File: src/memkind.c [hunk 1]
```diff
@@ -582,6 +582,9 @@ void memkind_init(memkind_t kind, bool check_numa)
             abort();
         }
     }
+    void *t = memkind_malloc(kind, 8);
```
Comment (maciekpac) [inline src/memkind.c:585]: What if it fails? E.g. DAX_KMEM returns null if no pmem is detected in the system, in such case, we would get segfault

**Review by maciekpac — CHANGES_REQUESTED** [review]
I'm not sure about the intent of this commit and the code looks dubious

**Review by bratpiorka — COMMENTED** [review]
I updated commit info and added a comment

> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/754#-:-Mw6HLWl6NM2EN0xSQl1:b-ekrsib)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @kilobyte, @maciekpac, and @PatKamin)

___
*[src/memkind.c, line 585 at r1](https://reviewable.io/reviews/memkind/memkind/754#-Mw2Z5bLcSKKw51pPJT2-r1-585:-Mw6HIt711RLJUq4GfeS:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/7048da5df4a7b04a8b754ae75a6d5a069c0d20c6/src/memkind.c#L585)):*
<details><summary><i>Previously, maciekpac wrote…</i></summary><blockquote>

What if it fails? E.g. DAX_KMEM returns null if no pmem is detected in the system, in such case, we would get segfault
</blockquote></details>

Done.

**Review by maciekpac — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/754)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @bratpiorka, @kilobyte, and @PatKamin)

___
*[src/memkind.c, line 585 at r1](https://reviewable.io/reviews/memkind/memkind/754#-Mw2Z5bLcSKKw51pPJT2-r1-585:-MwBBbCo5dOBEBoYG1n4:b-y6ers3) ([raw file](https://github.com/memkind/memkind/blob/7048da5df4a7b04a8b754ae75a6d5a069c0d20c6/src/memkind.c#L585)):*
<details><summary><i>Previously, bratpiorka (Rafał Rudnicki) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

The use case for this is really obscure - I would describe it in more detail - visibility in numactl, statistics, etc.

**Pull request closed without merging**